src/spdx/validation/package_validator.py

```diff
@@ -43,23 +44,18 @@ def validate_package_within_document(package: Package, document: Document) -> Li
     for message in validate_spdx_id(package.spdx_id, document):
         validation_messages.append(ValidationMessage(message, context))
 
-    # TODO: make test for this (https://github.com/spdx/tools-python/issues/386)
     if not package.files_analyzed:
-        package_contains_relationships = [relationship for relationship in document.relationships if
-                                          relationship.relationship_type == RelationshipType.CONTAINS and relationship.spdx_element_id == package.spdx_id]
-        if package_contains_relationships:
-            validation_messages.append(
-                ValidationMessage(
-                    f"package must contain no elements if files_analyzed is False, but found {package_contains_relationships}",
-                    context)
-            )
+        package_contains_relationships = filter_by_type_and_origin(document.relationships, RelationshipType.CONTAINS,
+                                                                   package.spdx_id)
+        contained_in_package_relationships = filter_by_type_and_target(document.relationships,
+                                                                       RelationshipType.CONTAINED_BY, package.spdx_id)
+
+        combined_relationships: List[Relationship] = package_contains_relationships + contained_in_package_relationships
 
-        contained_in_package_relationships = [relationship for relationship in document.relationships if
-                                              relationship.relationship_type == RelationshipType.CONTAINED_BY and relationship.related_spdx_element_id == package.spdx_id]
-        if contained_in_package_relationships:
+        if combined_relationships:
             validation_messages.append(
                 ValidationMessage(
-                    f"package must contain no elements if files_analyzed is False, but found {contained_in_package_relationships}",
+                    f"package must contain no elements if files_analyzed is False, but found {combined_relationships}",
```

Comment:
Reading this, I was wondering if a package must contain no elements in general or only files? The spec is mostly referring to files and I am not sure, if a package could contain anything other than a file.

Reply:
I don't think it can contain anything else than files. This is not explicitly stated by the spec for relationships, so who knows... I would leave it like that for now.